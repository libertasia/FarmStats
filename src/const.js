export const AppRoute = {
  ROOT: `/`,
};

export const APIRoute = {
  FARMS: `/v1/farms`,
};

export const TabTypes = {
  GRAPHS: `graphs`,
  TABLE: `table`,
};

export const TableHeaders = [
  {
    title: `Location`,
  },
  {
    title: `Datetime`,
  },
  {
    title: `Metric`,
  },
  {
    title: `Value`,
  }
];
