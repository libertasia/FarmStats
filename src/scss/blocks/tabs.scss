.tabs {
  margin-bottom: 50px;
}

.tabs__list {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.tabs__list-item--active {
  border: 2px solid $border-color;
  border-bottom-color: transparent;
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
}

.tabs__list-item--inactive {
  border-bottom: 2px solid $border-color;
}

.tabs__link-btn {
  background-color: transparent;
  border: none;
  cursor: pointer;
  text-decoration: none;
  display: inline;
  margin: 0;
  padding: 10px 15px;
  font-family: $main-font;
  font-weight: 500;
  font-size: 20px;
  letter-spacing: 0.1em;
  text-transform: capitalize;
  color: $light-blue;

  &:hover {
    transform: scale3d(1.02,1.02,1.02);
  }
}
