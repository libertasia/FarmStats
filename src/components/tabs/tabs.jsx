import React from "react";
import {connect} from 'react-redux';
import PropTypes from "prop-types";
import {TabTypes} from "../../const";
import {setActiveTab} from "../../store/action";
import {getActiveTab} from "../../store/selectors";

const TabDetails = [
  {
    type: TabTypes.TABLE,
    title: `Table`
  },
  {
    type: TabTypes.GRAPHS,
    title: `Graphs`
  },
];

const Tabs = (props) => {
  const {activeTab, onSetActiveTab} = props;

  const handleTabClick = (evt) => {
    evt.preventDefault();
    onSetActiveTab(evt.currentTarget.dataset.id);
  };

  return (
    <ul className="tabs-list container">
      {
        TabDetails.map((tab) =>
          (
            <li key={tab.type} data-id={tab.type} onClick={handleTabClick} className={`tabs-list__item ${tab.type === activeTab ? `tabs-list__item--active` : `tabs-list__item--inactive`}`}>
              <button type="button" className="tabs-list__link-btn">{tab.title}</button>
            </li>
          ))
      }
    </ul>
  );
};

Tabs.propTypes = {
  activeTab: PropTypes.string.isRequired,
  setActiveTab: PropTypes.func,
};

const mapStateToProps = (state) => ({
  activeTab: getActiveTab(state),
});

const mapDispatchToProps = (dispatch) => ({
  onSetActiveTab(tabId) {
    dispatch(setActiveTab(tabId));
  },
});

export {Tabs};
export default connect(mapStateToProps, mapDispatchToProps)(Tabs);
