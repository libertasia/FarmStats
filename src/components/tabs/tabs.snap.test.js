import React from "react";
import {Provider} from "react-redux";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {TabType} from "../../const";
import {Tabs} from "./tabs";

const mockStore = configureStore({});
const store = {
  VIEW: {
    activeTab: TabType.TABLE,
  }
};

it(`Tabs list renders correctly`, () => {
  const {container} = render(
      <Provider store={mockStore(store)}>
        <Tabs
          activeTab={TabType.TABLE}
          onSetActiveTab={jest.fn()}
        />
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
