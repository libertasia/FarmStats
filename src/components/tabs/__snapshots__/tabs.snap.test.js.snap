// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs list renders correctly 1`] = `
<div>
  <div
    class="tabs"
  >
    <ul
      class="tabs__list"
    >
      <li
        class="tabs__list-item tabs__list-item--active"
        data-id="table"
      >
        <button
          class="tabs__link-btn"
          type="button"
        >
          table
        </button>
      </li>
      <li
        class="tabs__list-item tabs__list-item--inactive"
        data-id="graphs"
      >
        <button
          class="tabs__link-btn"
          type="button"
        >
          graphs
        </button>
      </li>
    </ul>
  </div>
</div>
`;
